### DISCUSSION TITLE
What does LL stand for?

### DISCUSSION THREAD

**Discussion by bivainis** [ivanhofer/typesafe-i18n#528] [Q&A] [answered]
Help me settle the debate: is there something that `LL` stands for in the context of this library? I was trying to read through the readme files and codebase, and couldn't find it documented.

My guesses:
- Loaded Locales
- Localised Language
- Language Locale
- Language Library
- Locale List
- Locale Labels

**Comment by ivanhofer** [comment, marked as answer]
That's a good question 😅

In most libraries you would probably see a `t` translation functions.
But since `typesafe-i18n` can be [used in multiple ways](https://github.com/ivanhofer/typesafe-i18n/tree/main/packages/runtime#usage), I needed to have three different names so you would be able to immediately tell the difference bretween instances of `i18nString`, `i18nObject` and `i18n` in your code.
The first character that popped up in my mind was `l` (which probably stands for `locale`?).
Since it can be hard to tell the difference between 
- `l` (lovercase `L`) and
- `I` (uppercase `i`)

I did choose to use `L`, resulting into `L`, `LL` and `LLL` for the names of those instances.

**Reply by swernerx** [reply]
I think the naming is difficult. Feels a bit cumbersome and I wish we might provide more verbose but talking APIs in a future release. BTW: I also don't like the `t` of `i18next`.... especially in React Context `LL` seems unnecessarily short. Considering the barriers to entry and the possibilities of compression, I would prefer `msgs` for example....

**Reply by ivanhofer** [reply]
Hi @swernerx,
feel free to suggest a good naming structure for all 3 instances :)
`LL` is the default export of most modules (where possible) so you can name it whatever you like. And if you really dislike the `LL` you can always wrap it in a function and return a different naming schema.

**Comment by bivainis** [comment]
thanks for the in-depth answer, would've never guessed it, can finally find peace in my head :)
